src/groupByKey.ts

```diff
@@ -1,7 +1,24 @@
+/* eslint no-param-reassign: "error" */
+
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items:{} [], key: string): GroupsMap<{}> {
```

Comment:
давай більше можливостей тайпскрипта підключимо
```suggestion
export function groupByKey<T, K extends keyof T>(items: T[], key: K): GroupsMap<T> {
```

Reply:
тепер давай подумаєм як можна спростити подальший код?